// ng2-slim-loading-bar/__ivy_ngcc__/src/slim-loading-bar.component.js
// Copyright (C) 2016 Sergey Akopkokhyants
// This project is licensed under the terms of the MIT license.
// https://github.com/akserg/ng2-slim-loading-bar
import { Component, Input } from '@angular/core';
import { SlimLoadingBarService, SlimLoadingBarEventType } from './slim-loading-bar.service';
import { isPresent } from './slim-loading-bar.utils';
/**
 * A Slim Loading Bar component shows message loading progress bar on the top of web page or parent component.
 */
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from './slim-loading-bar.service';
export var SlimLoadingBarComponent = (function () {
    function SlimLoadingBarComponent(service) {
        this.service = service;
        this.progress = '0';
        this.color = 'firebrick';
        this.height = '2px';
        this.show = true;
    }
    SlimLoadingBarComponent.prototype.ngOnInit = function () {
        var _this = this;
        this.service.events.subscribe(function (event) {
            if (event.type === SlimLoadingBarEventType.PROGRESS && isPresent(event.value)) {
                _this.progress = event.value;
            }
            else if (event.type === SlimLoadingBarEventType.COLOR) {
                _this.color = event.value;
            }
            else if (event.type === SlimLoadingBarEventType.HEIGHT) {
                _this.height = event.value;
            }
            else if (event.type === SlimLoadingBarEventType.VISIBLE) {
                _this.show = event.value;
            }
        });
    };
    /** @nocollapse */
    SlimLoadingBarComponent.ctorParameters = function () { return [
        { type: SlimLoadingBarService, },
    ]; };
    SlimLoadingBarComponent.propDecorators = {
        'progress': [{ type: Input },],
        'color': [{ type: Input },],
        'height': [{ type: Input },],
        'show': [{ type: Input },],
    };
SlimLoadingBarComponent.ɵfac = function SlimLoadingBarComponent_Factory(t) { return new (t || SlimLoadingBarComponent)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.SlimLoadingBarService)); };
SlimLoadingBarComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: SlimLoadingBarComponent, selectors: [["ng2-slim-loading-bar"]], inputs: { progress: "progress", color: "color", height: "height", show: "show" }, decls: 2, vars: 10, consts: [[1, "slim-loading-bar"], [1, "slim-loading-bar-progress"]], template: function SlimLoadingBarComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵelement(1, "div", 1);
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵstyleProp("width", ctx.progress + "%")("background-color", ctx.color)("color", ctx.color)("height", ctx.height)("opacity", ctx.show ? "1" : "0");
    } }, encapsulation: 2 });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(SlimLoadingBarComponent, [{
        type: Component,
        args: [{
                selector: 'ng2-slim-loading-bar',
                template: "\n<div class=\"slim-loading-bar\">\n    <div class=\"slim-loading-bar-progress\" [style.width]=\"progress + '%'\" [style.backgroundColor]=\"color\" [style.color]=\"color\"\n        [style.height]=\"height\" [style.opacity]=\"show ? '1' : '0'\"></div>\n</div>"
            }]
    }], function () { return [{ type: ɵngcc1.SlimLoadingBarService }]; }, { progress: [{
            type: Input
        }], color: [{
            type: Input
        }], height: [{
            type: Input
        }], show: [{
            type: Input
        }] }); })();
    return SlimLoadingBarComponent;
}());
